code-snippets/StaticBuilder.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE CPP #-}
+
+import Reflex.Dom
+import Data.Text as T
+import Data.Text.Encoding as T
+import Data.ByteString.Char8 as BS8
+
+main = do
+#if defined (ghcjs_HOST_OS)
```

Comment:
Especially in learning materials, CPP should be strongly avoided. In this particular case, the frontend should be library-ized, so that the backend can simply import it and run it at runtime. Most apps will want to populate the statically rendered site with runtime content, so this will be necessary anyway.

Reply:
Yes.. ideally we should have two separate "top" modules to demonstrate this. I am actually not happy with the idea of a separate "code-snippets" altogether. I would like to add all the code here to reflex-examples (once that is fixed), so that we can build them and test.
 Ref https://github.com/reflex-frp/reflex-examples/issues/13

Reply:
This file will no longer be a part of the docs. Please ignore this